prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -182,6 +182,12 @@ class DetailedSubmissionHistory(
     @JsonIgnore
     var nextActionScheduled = false
 
+    /**
+     * Flag to check if there's a next action for a report related to this submission
+     */
+    @JsonIgnore
+    var hasNextAction = false
```

Comment:
I totally forget how the `nextActionScheduled` is used, but maybe they can get collapsed?